packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -122,6 +132,11 @@ const baseCases = [
     code: 'foo.bar && foo.bar?.() && foo.bar?.().baz',
     output: 'foo.bar?.()?.baz',
   },
+  // TODO: deepest left node already pre-optional chained
+  // {
+  //   code: 'foo?.bar && foo.bar?.() && foo.bar?.().baz',
+  //   output: 'foo?.bar?.()?.baz',
+  // },
```

Comment:
@JoshuaKGoldberg This TODO is a false-negative that I found in the existing `&&` matches, not related to the `!x ||` matches that I'm adding

Reply:
Oh? Is this a new issue that should be filed, then?

Reply:
I'll open one and see when I can attend to it